### PR TITLE
Fix some tests

### PR DESCRIPTION
**Pull request by dwalluck**
* Fix test exit codes
* Fix file separators
* Don't use shell glob when not in shell (always set outputFileName)
* Always set working directory of ProcessBuilder
* Don't call external sort command (use Groovy when sort is needed)

### CODE REVIEW

File: src/it/test19-intermediate-directories/verify.groovy [hunk 1]
```diff
@@ -1,7 +1,7 @@
 def verifyFileInodes() {
-    Process proc = ['rpm', '-q', '--queryformat', '[%{FILEINODES} %{FILEMODES:perms} %-13{FILEUSERNAME} %-14{FILEGROUPNAME} %{FILENAMES}\n]', basedir.path + '/target/*.rpm'].execute() | 'sort -n'.execute()
+    Process proc = ['rpm', '-q', '--queryformat', '[%{FILEINODES} %{FILEMODES:perms} %-13{FILEUSERNAME} %-14{FILEGROUPNAME} %{FILENAMES}\n]', basedir.toString().replace(File.separator, "/") + '/target/test19.rpm'].execute()
     proc.waitFor()
-    return proc.in.getText().trim()
+    return proc.in.getText().split().sort()
```
Comment (ctron) [inline src/it/test19-intermediate-directories/verify.groovy:4]: As the sort order seems important here, we should not change this.

Comment (dwalluck) [reply src/it/test19-intermediate-directories/verify.groovy:4]: It should just be a "better" way of keeping the same behavior as before.

I removed the call to `sort -n` (which only works in a shell) and replaced it with pure Groovy `split().sort()`, so

Comment (dwalluck) [reply src/it/test19-intermediate-directories/verify.groovy:4]: Note that this pull request currently causes some test failures, but this one passes.

I think that the test failures are correct now and show some issues with the packager code.

Comment (ctron) [reply src/it/test19-intermediate-directories/verify.groovy:4]: Ah sorry, I missed that.

**Review by ctron — DISMISSED** [review]

**Comment by dwalluck** [comment]
@ctron The tests pass with the exception of the test yum1. The yum1 test is using a file in the source tree (`src/it/yum1/src/main/resources/test1-1.0.0-0.201606241041-noarch.rpm`) as opposed to creating it with the latest code.

**Review by ctron — APPROVED** [review]

**Pull request merged**